nh_eobs_mental_health/models/nh_clinical_wardboard.py

```diff
@@ -324,3 +335,38 @@ def force_v7_api(cls, obj):
         """
         if '_ids' in obj.__dict__:
             obj.__dict__.pop('_ids')
+
+    # Acuity Board grouping
+    @api.model
+    def _get_acuity_groups(self, ids, domain, read_group_order=None,
+                           access_rights_uid=None,):
+        """
+        Override _get_cr_groups to include obs_stop and new patient
+        / restarted observations - EOBS-404
+
+        :param ids: record ids
+        :param domain: Domain to filter groups with
+        :param read_group_order: Order to read the groups in
+        :param access_rights_uid: User ID to use for access rights
+        :returns: Dictionary of groups and folded states
+        """
+        res = copy.deepcopy(self.acuity_selection)
+        fold = {r[0]: False for r in res}
```

Comment:
I copied and pasted this code from the get_cr_groups code on nh_eobs. I'll refactor but yes the dictionary keys is being created by iterating over the list items